runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -347,6 +337,76 @@ public void onPlayerMenuOptionClicked(PlayerMenuOptionClicked event)
 		}
 	}
 
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged gameStateChanged)
```

Comment:
Also, do all of this only if panel is enabled. All this fetching logic is pretty excessive to have it running when it is not needed

Reply:
This isn't actually fetching anything? It has to rebuild the UI to color the current world green.

Reply:
Yes